io-native/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/RecordChannelUtils.java

```diff
@@ -107,6 +112,23 @@ public static Object getNext(BObject channel) {
         }
     }
 
+    public static Object readRecord(BObject channel, BString separator) {
+        BufferedReader bufferedReader = (BufferedReader)
+                channel.getNativeData(BUFFERED_READER_ENTRY);
```

Comment:
Can this native data be null? If it is a scenario lets add null check

Reply:
BufferedInputStream can be null, so I added a null check there (https://github.com/ballerina-platform/module-ballerina-io/pull/31/commits/98acb67cdbfd15b787de2bc02c14a106f0615bfe). Since this BufferedReader initialized in the init function, this cannot be null. So, I removed the unnecessary null checks.